pkg/secretlock/local/local_secret_lock.go

```diff
@@ -0,0 +1,83 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package local
+
+import (
+	"crypto/cipher"
+	"encoding/base64"
+	"io"
+	"os"
+
+	"github.com/google/tink/go/subtle/random"
+	"golang.org/x/crypto/chacha20poly1305"
+
+	"github.com/hyperledger/aries-framework-go/pkg/secretlock"
+)
+
+// Lock is a secret lock service responsible for encrypting keys using a master key stored as environment variable
+type Lock struct {
+	aead cipher.AEAD
+}
+
+// NewService creates a new instance of a local secret lock service for a master key in path
+func NewService(path string) (secretlock.Service, error) {
+	masterKeyFile, err := os.OpenFile(path, os.O_RDONLY, 0600)
+	if err != nil {
+		return nil, err
+	}
+
+	mkData := make([]byte, 128)
+
+	n, err := io.ReadFull(masterKeyFile, mkData)
+	if err != io.ErrUnexpectedEOF {
+		return nil, err
+	}
+
+	mkData = mkData[0:n]
+
+	masterKey, err := base64.URLEncoding.DecodeString(string(mkData))
+	if err != nil {
+		return nil, err
+	}
+
+	aead, err := chacha20poly1305.NewX(masterKey)
```

Comment:
cannot hard-code to chacha.

why not AES GCM?

Reply:
Secret locks are not meant to be exposed to the user, it's an internal service (used by KMS primarily), so the user will not choose this algorithm.

chacha is used for faster/shorter crypto operations

Reply:
It's not about user choice, it's about ability to meet a NIST-only requirement - it should be possible to build an agent that only uses AES for key wrapping.

Reply:
switched